enhancements/authentication/pod-security-admission-autolabeling.md

```diff
@@ -0,0 +1,293 @@
+---
+title: pod-security-admission-autolabeling
+authors:
+  - "@stlaz"
+reviewers:
+  - "@deads2k"
+  - "@s-urbaniak"
+  - "@soltysh"
+approvers:
+  - "@sttts"
+  - "@mfojtik"
+api-approvers: # in case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers)
+  - @sttts"
+creation-date: 2022-02-04
+last-updated: 2022-02-04
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/AUTH-133
+see-also:
+  - "/enhancements/authentication/pod-security-admission.md"
+replaces: []
+superseded-by: []
+---
+
+# Pod Security Admission Autolabeling
+
+## Summary
+
+This enhancement expands the "PodSecurity admission in OpenShift". It introduces
+an opt-in mechanism that allows users to to keep their workloads running when
+Pod Security Admission plugin gets turned on.
+
+## Motivation
+
+We want to adhere with the upstream pod security standards for our workloads but
+we also want to provide our users access to the Security Context Constraints
+(hereinafter SCCs) API that they are already used to. However, each of these
+admission plugins works a bit differently and so there must be a middle-man
+that synchronizes the privileges SCCs provide into privileges that Pod Security
+admission (hereinafter PSa) understands.
+
+### Goals
+
+1. OpenShift clusters can run with the restricted Pod Security admission level by default
+without breaking pod controller workloads
+2. Privileged users can opt-in/opt-out a namespace to and from the Pod Security admission
+
+### Non-Goals
+
+1. Make sure that bare pods (pods w/o pod controllers) continue working
+
+## Proposal
+
+### User Stories
+
+1. I would like my workloads to be admitted based on the current pod security standards
+   with minimal effort or no effort at all
+
+### Pod Security Admission Label Synchronization
+
+This section explains how pod security admission works and later elaborates on how to use
+the pod security labeling in order to synchronize SCC permissions into pod security admission
+restriction levels.
+
+The SCC permissions to PSa levels transformation gets explained at the end of this section.
+
+#### Pod Security Admission Introduction
+
+Pod Security admission validates pods' security according to the upstream
+[pod security standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/)
+and distinguishes three different security levels:
+- privileged - most privileged mode, anything is allowed
+- baseline   -  minimally restrictive policy which prevents known privilege escalations
+- restricted - heavily restricted policy, following current Pod hardening best practices
+
+By default, there is a cluster-global configuration which enforces the configured policies
+on pods and known workloads.
```

Comment:
We should explicitely say that despite the possibility to override on a namespace level we do not (officially) allow overriding the cluster-global configuration which is defaulting to `restricted`.

Reply:
added to non-goals